rotkehlchen/chain/ethereum/modules/balancer/v1/decoder.py

```diff
@@ -0,0 +1,274 @@
+            if proxied_event['ds_address'] != event.counterparty:
+                continue
+
+            if proxied_event['type'] == BalancerV1EventTypes.JOIN:
+                if (
+                    event.event_type == HistoryEventType.RECEIVE and
+                    event.event_subtype == HistoryEventSubType.NONE
+                ):
+                    event.event_subtype = HistoryEventSubType.RECEIVE_WRAPPED
+                    event.counterparty = CPT_BALANCER_V1
+                    event.notes = f'Receive {event.balance.amount} {token.symbol} from a Balancer v1 pool'  # noqa: E501
+                    return True
+                if (
+                    event.event_type == HistoryEventType.SPEND and
+                    event.event_subtype == HistoryEventSubType.NONE
+                ):
+                    event.event_type = HistoryEventType.DEPOSIT
+                    event.event_subtype = HistoryEventSubType.DEPOSIT_ASSET
+                    event.counterparty = CPT_BALANCER_V1
+                    event.notes = f'Deposit {event.balance.amount} {token.symbol} to a Balancer v1 pool'  # noqa: E501
+                    return True
+            elif proxied_event['type'] == BalancerV1EventTypes.EXIT:
+                if (
+                    event.event_type == HistoryEventType.RECEIVE and
+                    event.event_subtype == HistoryEventSubType.NONE
+                ):
+                    event.event_type = HistoryEventType.WITHDRAWAL
+                    event.event_subtype = HistoryEventSubType.REMOVE_ASSET
+                    event.counterparty = CPT_BALANCER_V1
+                    event.notes = f'Receive {event.balance.amount} {token.symbol} after removing liquidity from a Balancer v1 pool'  # noqa: E501
+                    return True
+                if (
+                    event.event_type == HistoryEventType.SPEND and
+                    event.event_subtype == HistoryEventSubType.NONE
+                ):
+                    event.event_subtype = HistoryEventSubType.RETURN_WRAPPED
+                    event.counterparty = CPT_BALANCER_V1
+                    event.notes = f'Return {event.balance.amount} {token.symbol} to a Balancer v1 pool'  # noqa: E501
+                    return True
+
+        return False
+
+    def _check_refunds_v1(
+            self,
+            transaction: EvmTransaction,  # pylint: disable=unused-argument
+            decoded_events: list[HistoryBaseEntry],
+            all_logs: list[EvmTxReceiptLog],  # pylint: disable=unused-argument
+    ) -> list[HistoryBaseEntry]:
+        """
+        It can happen that after sending tokens to the DSProxy in balancer V1 the amount of tokens
+        required for the deposit is lower than the amount sent and then those tokens are returned
+        to the DSProxy and then to the user.
+        """
+        deposited_assets = set()
+
+        for event in decoded_events:
+            if event.counterparty != CPT_BALANCER_V1:
```

Comment:
No this is not good enough. So we need more work on the post-decoding rules. I think we need to tag transactions with which protocols they decoded and then only go in the post-decoding rules if they match. Otherwise this system won't scale.

Reply:
I have added a new commit that modifies how the rules are used. Now this logic is only executed when there is a balancer event detected (as it should)